v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compiler/compiler.ts

```diff
@@ -177,8 +176,7 @@ export class NativeCompiler implements Compiler {
         args.push("--no-import-callback");
       } else if (semver.gte(this.version, "0.6.9")) {
         // version >= 0.6.9
-        const tmpFolder = path.join(os.tmpdir(), "hardhat-solc");
-        await mkdir(tmpFolder);
+        const tmpFolder = await mkdtemp("hardhat-solc-");
```

Comment:
I also did this here because the assumption here is that `tmpFolder` is empty, but the other part of the code is creating directories under `/tmp/hardhat-solc`. While unlikely to be a problem in practice, I still think it's better to just use a unique random empty directory each time.